client/history/client.go

```diff
@@ -1134,8 +1061,6 @@ func (c *clientImpl) RespondCrossClusterTasksCompleted(
 	var response *types.RespondCrossClusterTasksCompletedResponse
 	op := func(ctx context.Context, peer string) error {
 		var err error
-		ctx, cancel := c.createContext(ctx)
-		defer cancel()
 		response, err = c.client.RespondCrossClusterTasksCompleted(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
 		return err
 	}
```

Comment:
unless I'm misreading things, this is _very_ different from the now-generated code.

it may be an improvement (createContext is rather terrifying iirc) but it's not clear this is correct.

Reply:
I didn't get what is the concern about correctness?

Reply:
I think for this method it's correct. What's incorrect is these methods below don't have timeout but now were generated incorrectly with timeout. I'll fix it in the next pr. 
```
GetReplicationMessages
GetDLQReplicationMessages
CountDLQMessages
ReadDLQMessages
PurgeDLQMessages
MergeDLQMessages
GetCrossClusterTasks
GetFailoverInfo
```

Reply:
fixed in https://github.com/uber/cadence/pull/5737

Reply:
previously this was potentially modifying the context on each `op` call during (possibly multiple) request redirection retries.

now it's only doing that once, before calling the retry impl.  (well before that, as it's the outer-most layer currently)

that's not necessarily a small change.

Reply:
timeout wrapper is still inside the "redirect client". So I think it's the same. 
Basically 
```
ctx, cancel := c.createContext(ctx)
defer cancel()
response, err = c.client.RespondCrossClusterTasksCompleted(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
```
is modified to 
```
response, err = c.wrappedClient.RespondCrossClusterTasksCompleted(ctx, request, append(opts, yarpc.WithShardKey(peer))...)
```